runelite-client/src/main/java/net/runelite/client/externalplugins/ExternalPluginManager.java

```diff
@@ -189,6 +192,7 @@ private void refreshPlugins()
 					{
 						externalPlugins.add(manifest);
 
+						manifest.getJarFile().setLastModified(now);
```

Comment:
This only needs to happen when adding to `keep` right - otherwise the file is modified from the download which does a write()

Reply:
doing it always makes a race less likely

Reply:
That's fine as long as `setLastModified` fails okay if the file doesn't exist.

Reply:
as documented, it returns false and does not throw